### PR TITLE
Follow up of pr28

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/spoke/controllers/finalizercontroller/manifestwork_finalize_controller.go [hunk 1]
```diff
@@ -86,15 +83,15 @@ func (m *ManifestWorkFinalizeController) sync(ctx context.Context, controllerCon
 		return nil
 	}
 
-	appliedManifestWorkPresent, err := m.ifAppliedManifestWorkPresent(ctx, appliedManifestWorkName)
-	if err != nil {
+	_, err = m.appliedManifestWorkLister.Get(appliedManifestWorkName)
+	switch {
```
Comment (deads2k) [inline pkg/spoke/controllers/finalizercontroller/manifestwork_finalize_controller.go:87]: optional nit: 

```go
switch{
case errors.IsNotFound(err):
   // if the instance is not found, then we simply continue below this block to remove the finalizer
case err != nil:
  // return all unexpected errors
  return err
default:
  // appliedmanifestwork still exists, requeue the manifestwork to check in the next loop.
}
```

It keeps each condition small and tests every state of err.  It's an optional change.

**Comment by deads2k** [comment]
optional suggestion here: https://github.com/open-cluster-management/work/pull/36/files#r457338622

Up your discretion whether to take it.

/lgtm
/hold

holding in case you want to take the suggestion.

File: pkg/spoke/controllers/finalizercontroller/manifestwork_finalize_controller.go (hunk 1 above)
Comment (qiujian16) [reply pkg/spoke/controllers/finalizercontroller/manifestwork_finalize_controller.go:87]: yes, it is much clearer. Updated

**Pull request merged**